src/database.js

```diff
@@ -44,17 +47,31 @@ export class Database {
   }
 
   factoryFor(collectionName) {
-    return this.factories[collectionName];
+    const factory = this.factories[collectionName];
+
+    return factory ? factory.factory : undefined;
+  }
+
+  serializerFor(collectionName) {
+    const factory = this.factories[collectionName];
+
+    return factory ? factory.serializer : () => {}; 
```

Comment:
Actually I was thinking about it and maybe it has to be `null` that way it will fail instead of having some weird behaviours in the future...